### PR TITLE
[REVIEW] Add COUNT aggregation that includes null values

### PR DESCRIPTION
**Pull request by karthikeyann**
closes https://github.com/rapidsai/cudf/issues/3708
- [x] change COUNT to COUNT_VALID (groupby, rolling)
- [x] add COUNT_ALL Operator support
- [x] add unit tests for COUNT_ALL
- [x] update documentation (COUNT -> COUNT_VALID, COUNT_ALL)

### CODE REVIEW

File: cpp/include/cudf/aggregation.hpp [hunk 1]
```diff
@@ -61,8 +61,12 @@ std::unique_ptr<aggregation> make_min_aggregation();
 /// Factory to create a MAX aggregation
 std::unique_ptr<aggregation> make_max_aggregation();
 
-/// Factory to create a COUNT aggregation
-std::unique_ptr<aggregation> make_count_aggregation();
+/**
+ * @brief Factory to create a COUNT aggregation
+ * 
+ * @param _include_nulls include or exclude nulls flag
```
Comment (jrhemstad) [inline cpp/include/cudf/aggregation.hpp]:
```suggestion
 * @param _include_nulls Indicates if null values will be counted. 
```

File: cpp/src/aggregation/aggregation.cpp [hunk 2]
```diff
@@ -40,8 +40,12 @@ std::unique_ptr<aggregation> make_max_aggregation() {
   return std::make_unique<aggregation>(aggregation::MAX);
 }
 /// Factory to create a COUNT aggregation
-std::unique_ptr<aggregation> make_count_aggregation() {
-  return std::make_unique<aggregation>(aggregation::COUNT);
+std::unique_ptr<aggregation>
+make_count_aggregation(include_nulls _include_nulls) {
+  if(_include_nulls == include_nulls::NO)
```
Comment (jrhemstad) [inline cpp/src/aggregation/aggregation.cpp]:
```suggestion
   auto kind = (_include_nulls == include_nulls::YES) ? COUNT_ALL : COUNT_VALID;
   return std::make_unique<aggregation>(kind);
```

**Comment by karthikeyann** [comment]
@devavret would computing COUNT_ALL using group_offsets be faster? (I suppose group_offsets is computed before group_labels).

**Comment by devavret** [comment]
Unable to review properly because of ongoing github issues. Will try again in some time.

**Comment by devavret** [comment]
> @devavret would computing COUNT_ALL using group_offsets be faster? (I suppose group_offsets is computed before group_labels).

Yes. That's true. Even though practically always you'd find `group_labels` pre-computed but the thrust operation required is also cheaper if you use `group_offsets`.

**Comment by harrism** [comment]
rerun tests

**Comment by karthikeyann** [comment]
@jrhemstad for re-review

**Pull request merged**